README.md

```diff
@@ -29,11 +29,88 @@ composer require alleyinteractive/wp-rest-api-guard
 
 ## Usage
 
-Activate the plugin in WordPress and use it like so:
+The WordPress REST API is generally very public and can share a good deal of
+information with the internet anonymously. This plugin aims to make it easier to
+restrict access to the REST API for your WordPress site.
+
+### Settings Page
+
+The plugin can be configured via the Settings page (`Settings -> REST API
+Guard`) or via the relevant filter:
+
+<img width="1276" alt="Screen Shot 2022-10-06 at 4 24 01 PM" src="https://user-images.githubusercontent.com/346399/194411352-aa05e939-3fd1-4e37-a3d5-276c1c5c288f.png">
+
+### Preventing Access to User Information (`wp/v2/users`)
+
+By default, the plugin will restrict anonymous access to the users endpoint.
+This can be prevented in the plugin's settings or via code:
+
+```php
+add_filter( 'rest_api_guard_allow_user_access', fn () => true );
+```
+
+### Preventing Access to Index (`/`) or Namespace Endpoints (`wp/v2`)
+
+To prevent anonymous users from browing your site and discovering what plugins/post types are setup, the plugin restricts access to the index (`/`) and namespace (`wp/v2`) endpoints. This can be prevented in the plugin's settings or via code:
+
+```php
+// Allow index access.
+add_filter( 'rest_api_guard_allow_index_access', fn () => true );
+
+// Allow namespace access.
+add_filter( 'rest_api_guard_allow_namespace_access', fn ( string $namespace ) => true );
+```
+
+### Restrict Anonymous Access to the REST API
+
+The plugin can restrict anonymous access for any request to the REST API in the plugin's settings or via code:
+
+```php
+add_filter( 'rest_api_guard_prevent_anonymous_access', fn () => true );
+```
+
+### Limit Anonymous Access to Specific Namespaces/Routes (Allowlist)
+
+Anonymous users can be granted access only to specific namespaces/routes.
+Requests outside of these paths will be denied. This can be configured in the
+plugin's settings or via code:
+
+```php
+add_filter(
+	'rest_api_guard_anonymous_requests_allowlist',
+	function ( array $paths, WP_REST_Request $request ) {
```

Comment:
Since there is typehinting for parameters, recommend adding it for return type as well:
```suggestion
	function ( array $paths, WP_REST_Request $request ): array {
```